src/openloco/windows/CompanyList.cpp

```diff
@@ -1,18 +1,1732 @@
+                return cursor_id::hand_pointer;
+
+            return fallback;
+        }
+
+        // 0x00435D07
+        static void prepare_draw(window* self)
+        {
+            common::prepare_draw(self);
+
+            self->widgets[widx::scrollview].right = self->width - 4;
+            self->widgets[widx::scrollview].bottom = self->height - 14;
+
+            // Reposition header buttons
+            self->widgets[widx::sort_name].right = std::min(178, self->width - 8);
+
+            self->widgets[widx::sort_status].left = std::min(179, self->width - 8);
+            self->widgets[widx::sort_status].right = std::min(388, self->width - 8);
+
+            self->widgets[widx::sort_performance].left = std::min(389, self->width - 8);
+            self->widgets[widx::sort_performance].right = std::min(533, self->width - 8);
+
+            self->widgets[widx::sort_value].left = std::min(534, self->width - 8);
+            self->widgets[widx::sort_value].right = std::min(633, self->width - 8);
+
+            // Set header button captions
+            self->widgets[widx::sort_name].text = self->sort_mode == SortMode::Name ? string_ids::table_header_company_name_desc : string_ids::table_header_company_name;
+            self->widgets[widx::sort_status].text = self->sort_mode == SortMode::Status ? string_ids::table_header_company_status_desc : string_ids::table_header_company_status;
+            self->widgets[widx::sort_performance].text = self->sort_mode == SortMode::Performance ? string_ids::table_header_company_performance_desc : string_ids::table_header_company_performance;
+            self->widgets[widx::sort_value].text = self->sort_mode == SortMode::Value ? string_ids::table_header_company_value_desc : string_ids::table_header_company_value;
+        }
+
+        // 0x00435E56
+        static void draw(window* self, gfx::drawpixelinfo_t* dpi)
+        {
+            self->draw(dpi);
+            common::drawTabs(self, dpi);
+
+            auto args = FormatArguments();
+            if (self->var_83C == 1)
+                args.push(string_ids::company_singular);
+            else
+                args.push(string_ids::companies_plural);
+
+            args.push(self->var_83C);
+
+            auto xPos = self->x + 3;
+            auto yPos = self->y + self->height - 13;
+            gfx::draw_string_494B3F(*dpi, xPos, yPos, colour::black, string_ids::black_stringid, &args);
+        }
+
+        // 0x00435EA7
+        static void draw_scroll(window* self, gfx::drawpixelinfo_t* dpi, uint32_t scrollIndex)
+        {
+            //registers regs;
+            //regs.esi = (int32_t)self;
+            //regs.edi = (int32_t)dpi;
+            //call(0x00435EA7, regs);
+
+            auto colour = colour::get_shade(self->colours[1], 3);
+            gfx::clear(*dpi, colour * 0x01010101);
+
+            auto y = 0;
+            for (auto i = 0; i < self->var_83C; i++)
+            {
+                auto yPos = y + 25;
```

Comment:
yPos represents the top of the row and y represents the bottom. Rename the variables to make this easier to follow.